Sources/Basics/Lock.swift

```diff
@@ -0,0 +1,20 @@
+/*
```

Comment:
one small change: rename the file to `NSLock+Extensions.swift` and add it to the CMakeLists.txt